crates/router/src/core/payments.rs

```diff
@@ -3868,6 +4039,21 @@ pub fn is_network_transaction_id_flow(
         && payment_method_info.network_transaction_id.is_some()
 }
 
+pub fn is_network_token_with_network_transaction_id_flow(
+    is_connector_agnostic_mit_enabled: Option<bool>,
+    is_network_tokenization_enabled: bool,
+    payment_method_info: &storage::PaymentMethod,
+) -> bool {
+    is_connector_agnostic_mit_enabled == Some(true)
+        && is_network_tokenization_enabled
+        && payment_method_info.payment_method == Some(storage_enums::PaymentMethod::Card)
+        && payment_method_info.network_transaction_id.is_some()
```

Comment:
Instead bool send, Id's back so the in other place again and again get don't want to do get_required_value